packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -20,7 +21,7 @@ abstract class FlutterCommand extends Command {
   FlutterCommandRunner get runner => super.runner;
 
   /// Whether this command needs to be run from the root of a project.
-  bool get requiresProjectRoot => true;
+  bool get requiresProjectRoot => !_targetSpecified;
```

Comment:
This changes the semantic meaning of the call a bit. Perhaps add a new getter to `FlutterCommand` , something like `bool get requiresTarget => false;`, and verify that in `FlutterCommand._run()`? It would operate similar to our existing `requiresDevice` check. Subclasses that required a target would then need to override that getter to return true.